### ISSUE TITLE
Training Time

### ISSUE THREAD

**Issue by YiBenjing456** [CPJKU/score_following_game#10]
Hi authors, would you like to tell me how many timesteps did you use in the paper? I remember the paper report 6 hours for the training of A2C. Did you use max_steps = 15*1e9?

**Comment by fhenkel** [comment]
Hi, the maximum number of update steps is set to 1e9. In practice this number is never reached, as we apply early stopping.
The overall training time of course depends on your system as well. In our case it took several days to train an agent.

**Comment by YiBenjing456** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Thank you very much! One more question I'd like to ask is why the stp is 9k? Does it means only 9k steps were used for training. But the used steps in the outputs during the training process seem much more than 9k. Could you please help?
        
    
        
            
        
    
    
     
    
        
                    
            
                
                    
                            
                                
                            
                            
                                ldytongji0412
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 

    On 8/24/2021 04:35，Florian ***@***.***> wrote： 



Hi, the maximum number of update steps is set to 1e9. In practice this number is never reached, as we apply early stopping.
The overall training time of course depends on your system as well. In our case it took several days to train an agent.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by YiBenjing456** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    

    font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Thank you very much! One more question I'd like to ask is why the stp is 9k? Does it means only 9k steps were used for training. But the used steps in the outputs during the training process seem much more than 9k. Could you please help?
        The average reward is shown in the attachment.
    
        
            
        
    
    
     
    
        
                    
            
                
                    
                            
                                
                            
                            
                                ldytongji0412
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 

    On 8/24/2021 04:35，Florian ***@***.***> wrote： 



Hi, the maximum number of update steps is set to 1e9. In practice this number is never reached, as we apply early stopping.
The overall training time of course depends on your system as well. In our case it took several days to train an agent.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fhenkel** [comment]
The maximum training steps is not 9k but 1e9, which is arbitrarily chosen. This is just set to a very big number in order to not stop training too early.

**Comment by YiBenjing456** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Thanks for your reply. My question is that the x-axis of the average reward in the training results is '0 - 9k' as shown in the attached figure. I'm a little bit confused about what does the '9k' mean? Would you please like to help?
        Best regards!
    
        
            
        
    
    
     
    
        
                    
            
                
                    
                            
                                
                            
                            
                                ldytongji0412
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 

    On 8/25/2021 07:52，Florian ***@***.***> wrote： 



The maximum training steps is not 9k but 1e9, which is arbitrarily chosen. This is just set to a very big number in order to not stop training too early.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fhenkel** [comment]
Oh sorry for the misunderstanding. I think there is something wrong with the attachment as I can't see it.

**Comment by YiBenjing456** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        All right. Please see the attached pdf file.
        Best regards!
    
        
            
        
    
    
     
    
        
                    
            
                
                    
                            
                                
                            
                            
                                ldytongji0412
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 

    On 8/26/2021 03:23，Florian ***@***.***> wrote： 



Oh sorry for the misunderstanding. I think there is something wrong with the attachment as I can't see it.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fhenkel** [comment]
I still think that there is something wrong with the attachment. Could you please check if you can see it in the issue thread.
Otherwise you can also drop me a mail (see the paper for the address).
